lib/src/whereami.cc

```diff
@@ -1,16 +1,30 @@
 #include <whereami/version.h>
 #include <whereami/whereami.hpp>
-
 #include <leatherman/logging/logging.hpp>
+#include <internal/sources/dmi_source.hpp>
+#include <internal/detectors/virtualbox_detector.hpp>
 
-namespace whereami {
+using namespace std;
 
-    using namespace std;
+namespace whereami {
 
     string version()
     {
         LOG_DEBUG("whereami version is {1}", WHEREAMI_VERSION_WITH_COMMIT);
         return WHEREAMI_VERSION_WITH_COMMIT;
     }
 
-}  // whereami
+    vector<string> hypervisors()
+    {
+        vector<string> result;
+        sources::dmi dmi_source;
+
+        auto virtualbox_result = detectors::virtualbox(dmi_source);
+
+        if (virtualbox_result) {
+            result.emplace_back(string{vm::virtualbox});
```

Comment:
I don't think you need the manual `string` constructor here - since `emplace_back` forwards its arguments directly to the underlying class's constructor, you should be able to just use `emplace_back(vm::virtualbox)`.